src/HotChocolate/Core/src/Types/Types/FieldCollection.cs

```diff
@@ -24,10 +24,12 @@ public FieldCollection(IEnumerable<T> fields, bool sortByName = false)
             }
 
             _fields = sortByName
-                ? fields.OrderBy(t => t.Name).ToList()
-                : fields is List<T> list ? list : fields.ToList();
+                ? fields.OrderBy(t => t.Name).ToArray()
+                : fields is T[] list ? list : fields.ToArray();
+
+            _fieldsLookup = new Dictionary<NameString, (int Index, T Field)>(_fields.Length);
```

Comment:
Is this a good idea?

Reply:
This is the initial capacity 
the dict can always expand.